### DISCUSSION TITLE
Add compatibility with Azure's OpenAI API Endpoints

### DISCUSSION THREAD

**Discussion by talk2MeGooseman** [mgallo/openai.ex#46] [Ideas]
### Describe the feature or improvement you're requesting

Would you also be willing to have a setting to make this library compatible with Azure's version of OpenAI API endpoints? 

This would mirror openai library for Python https://github.com/openai/openai-python#microsoft-azure-endpoints
Azure only uses a subset of the endpoints OpenAI provides with a different request URL. 

Here is a link to the Swagger doc for the endpoints for auditing if feasible. I am also willing to help add this feature.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/stable/2022-12-01/inference.json

**Comment by mgallo** [comment]
hey, thanks for the suggestion. really interesting. I'll have a look at their implementation and see what can be done. I'm a bit busy these days, but I'll look into it as soon as I can 👀

**Reply by ArthV** [reply]
Do we have an update about this?
